preliz/internal/optimization.py

```diff
@@ -227,12 +227,15 @@ def optimize_beta_mode(lower, upper, tau_not, mode, dist, mass, prob):
             tau_not += 0.5 * tau_not
 
 
-def optimize_pymc_model(fmodel, target, draws, prior, initial_guess, bounds, var_info, p_model):
+def optimize_pymc_model(fmodel, targets, draws, prior, initial_guess, bounds, var_info, p_model):
     for _ in range(400):
         # can we sample systematically from these and less random?
         # This should be more flexible and allow other targets than just
         # a preliz distribution
-        obs = target.rvs(draws, random_state=np.random.default_rng(_))
+        if isinstance(targets, list):
+            obs = get_weighted_rvs([d[0] for d in targets], [w[1] for w in targets], draws, _)
```

Comment:
better to pass target to this function and extract the info internally

Reply:
Done